### DISCUSSION TITLE
when helper will inject to result code

### DISCUSSION THREAD

**Discussion by IronLu233** [swc-project/swc#1925] [Q&A] [answered]
I'm trying fixing #1924.

code with generator will call macro `helper!(interop_require_default, "interopRequireDefault")` in 
https://github.com/IronLu233/swc/blob/104be9837b39bfd88a5ce4fcb47a7bd8915cc993/ecmascript/transforms/module/src/common_js.rs#L576

But `_interopRequireDefault` implements is not existed in the result.

When I import the other thing using import syntax, `_interopRequireDefault` is injected correctlly.

I try to excute `enable_helper!(interop_require_default);` before `helper!(interop_require_default, "interopRequireDefault")` , but it's not work.

So when helper will inject to result code, and when is not?

**Comment by kdy1** [comment, marked as answer]
Helper is injected by `inject_helpers` pass (https://rustdoc.swc.rs/swc_ecma_transforms_base/helpers/fn.inject_helpers.html) and it's chained like https://github.com/swc-project/swc/blob/7a8ad8826a0072a53993977e68b00ccabe525d01/src/builder.rs#L199
